src/components/pages/devices/devices.js

```diff
@@ -1,14 +1,30 @@
 // Copyright (c) Microsoft. All rights reserved.
 
 import React, { Component } from 'react';
+import { Btn } from 'components/shared';
 
 import './devices.css';
 
 export class Devices extends Component {
-  render () {
+  constructor(props) {
+    super(props);
+    if (props.initialize) {
+      this.props.initialize();
+    }
+  }
+
+  render() {
+    const { t, devices, refresh } = this.props;
+    var listItems = t('components.devices.noneFound');
+    if (devices != null && devices.length > 0) {
+      listItems = devices.map((device, index) =>
+        <li key={index} className="device">{device.id}</li>
+      );
+    }
     return (
       <div className="devices-container">
-        Devices Page
+        <ul>{listItems}</ul>
```

Comment:
I wouldn't do it like this since listItems is now heterogeneous variable. That causes 1) a conflict in the naming (e.g. listItems might not be a list at all) and 2) could result in `ul` tags with just a string inside them, which is odd html. A better approach would be something like:

```
renderDevices(devices, noDevicesMessage) {
    if (devices && devices.length) {
        return render (
          <ul>
            { devices.map((device, index) => <li key={index} className="device">{device.id}</li>) }
          </ul>
        );
    }
    return noDevicesMessage
}
...
// in the render
{ this.renderDevices(devices, t('components.devices.noneFound'))}
```
Of some variation with the logic embedded in the jsx.